pandas/tests/groupby/test_apply.py

```diff
@@ -851,3 +851,39 @@ def test_apply_function_returns_non_pandas_non_scalar(function, expected_values)
     result = df.groupby("groups").apply(function)
     expected = pd.Series(expected_values, index=pd.Index(["A", "B"], name="groups"))
     tm.assert_series_equal(result, expected)
+
+
+def test_apply_fast_slow_identical():
+    # GH 31613
+
+    df = DataFrame({"A": [0, 0, 1], "b": range(3)})
+
+    # For simple index structures we check for fast/slow apply using
+    # an identity check on in/output
+    def slow(group):
+        return group
+
+    def fast(group):
+        return group.copy()
+
+    fast_df = df.groupby("A").apply(fast)
+    slow_df = df.groupby("A").apply(slow)
+
+    tm.assert_frame_equal(fast_df, slow_df)
+
+
+def test_gh14927():
+    # GH 14927
+    df = pd.DataFrame({"g": [1, 2, 2, 2], "a": [1, 2, 3, 4], "b": [5, 6, 7, 8]})
+
+    df1 = df.groupby("g").apply(lambda x: x)
```

Comment:
either parameterize this test, or call one of these expected and do 

result =
assert_frame_equal

result = ....

...
and so on

Reply:
done